kitsune/search/v2/base.py

```diff
@@ -79,6 +83,12 @@ def prepare(cls, instance, parent_id=None, **kwargs):
         if parent_id:
             obj.meta.id = parent_id
 
+        # check if the instance is suitable for indexing
+        # the prepare method of each Document Type can mark an object
+        # not suitable for indexing based on criteria defined on each said method
+        if hasattr(instance, "es_discard_doc"):
+            obj.es_discard_doc = True
```

Comment:
The use of `hasattr` and `True` throughout makes me a bit uncomfortable - IMO `True` implies setting `instance.es_discard_doc = False` will keep the doc (when it'll actually delete it) - and the same applies to the `es_discard_doc` attribute on the document obj.

Could we change to either using `getattr(x, "es_discard_doc", False)` or `x.es_discard_doc = None` (I believe that's the pythonic way of initializing an empty attribute)?

Reply:
I get what you are saying but I am trying to find an nice way to update this but the `hasattr` has nothing to do with the value of the attr. It just checks whether an attr exists. This is more suitable compared to getattr which will return the default value eg False if the attr is not found but it will return the value of the attr if found which is more prone to errors.

Reply:
We could add a helper function like:
```python
def es_discard_doc(instance_or_obj):
    return bool(getattr(instance_or_obj, "es_discard_doc", False))
```

Reply:
I still believe that `hasattr` is a better fit. We care about the existence of an attribute and not it's value. Some documents will have it, others will not.

With the snippet above, the `getattr` will return either False b/c the attr is missing or the actual value of the attribute. The function will end up casting to bool and check either for the existence of an attribute or return the actual value. These are different things. It's going to work just fine because of the cast and the use of booleans though I don't see the reason to exchange the hasattr with getattr and force a comparison of different things.

Reply:
> We care about the existence of an attribute and not it's value. Some documents will have it, others will not.

If this is the case, then we shouldn't be doing:
```
instance.es_discard_doc = True
```
Because that then implies that setting:
```
instance.es_discard_doc = False
```
won't discard the doc, when it still will.

Happy to keep `hasattr`, but then we should do:
```
instance.es_discard_doc = None
```
to make it clear that it's the existence of the attribute, not it's value, which controls whether a doc is discarded.

Reply:
Oh, I see you've set it to `"unindex_me"`, that'll work.